modules/slugs/admin/slugs-migrate-qts.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * Legacy meta and options from QTS plugin.
+ */
+const QTX_SLUGS_LEGACY_QTS_META_PREFIX    = '_qts_slug_';
+const QTX_SLUGS_LEGACY_QTS_OPTIONS_PREFIX = '_qts_';
+const QTX_SLUGS_LEGACY_QTS_OPTIONS_NAME   = 'qts_options';
+
+/**
+ * Check if slugs meta should be migrated from the legacy QTS postmeta and termmeta.
+ *
+ * @return string messages giving details, empty if new meta found or no legacy meta found.
+ */
+function qtranxf_slugs_check_migrate_qts() {
+    global $wpdb;
+
+    /**
+     * Generic function that counts the slugs meta, legacy (QTS) or new (QTX).
+     *
+     * @param string $table name of the meta table (postmeta, termmeta)
+     * @param string $prefix prefix for the meta key
+     * @param string[] $msg array of messages, updated
+     *
+     * @return void
+     */
+    $count_slugs = function ( $table, $prefix, &$msg ) use ( $wpdb ) {
+        $results = $wpdb->get_var( "SELECT count(*) FROM  $table WHERE meta_key like '$prefix%'" );
+        if ( $results ) {
+            $msg[] = sprintf( __( "Found %s slugs from $table.", 'qtranslate' ), $results );
+        }
+    };
+
+    $msg = [];
+    $count_slugs( $wpdb->postmeta, QTX_SLUGS_META_PREFIX, $msg );
+    $count_slugs( $wpdb->termmeta, QTX_SLUGS_META_PREFIX, $msg );
+    if ( ! empty( $msg ) ) {
+        // Found some post/term meta with the new keys, no migrate to suggest (it can still be done manually).
+        return '';
+    }
+
+    $msg = [];
+    $count_slugs( $wpdb->postmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+    $count_slugs( $wpdb->termmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+
+    return empty ( $msg ) ? $msg : implode( '<br>', $msg );
+}
+
+/**
+ * Migrate slugs meta by migrating the legacy QTS postmeta and termmeta to QTX.
+ * Attention: current slugs meta are deleted.
+ *
+ * @param bool $db_commit true to commit changes, false for dry-run mode.
+ *
+ * @return string messages giving details.
+ */
+function qtranxf_slugs_migrate_qts_meta( $db_commit ) {
+    global $wpdb;
+
+    $new_prefix = QTX_SLUGS_META_PREFIX;
+    $old_prefix = QTX_SLUGS_LEGACY_QTS_META_PREFIX;
+
+    /**
+     * Generic function that migrates QTS meta to QTX meta.
+     *
+     * @param string $table name of the meta table (postmeta, termmeta)
+     * @param string[] $msg array of messages, updated
+     *
+     * @return void
+     */
+    $migrate_meta = function ( $table, &$msg ) use ( $wpdb, $old_prefix, $new_prefix ) {
+        $results = $wpdb->query( "DELETE FROM $table WHERE meta_key like '$new_prefix%'" );
```

Comment:
I think we should count here the number of `QTX_SLUGS_LEGACY_QTS_META_PREFIX` entries and return if it's 0.
It is pointless to remove current data if nothing is there to be imported, it is likely a mistaken operation, and out of function intent anyways.

Reply:
It's tricky to count before the DELETE is done. The rename must be done after the DELETE, otherwise there might be some conflicts with the key names. Some keys may already exist, some not. The admin should use the dry-run mode to see the effects before commit.

Reply:
But perhaps it doesn't matter to know if all will be migrated. If there's no QTS meta at all, we can assume we don't want to delete anything, it's fine. I will push a change.

Reply:
Add pre-check QTS for conditional delete and migrate by table. It's safer now.

Reply:
In fact I made a confusion with the previous conservative import. In the new case it's much simpler, it becomes all or nothing. So it makes perfectly sense to count QTS data before deletion in QTX. Good point!